### PR TITLE
Integrate poetry, nox, release drafter

### PR DESCRIPTION
**Pull request by Imipenem**
**Associated Template/Command/Core**

* Move from `pip` to `poetry` build system
* Add `nox`
* Add release drafter to replace Changelog

Cookietemple core

**PR Checklist**

 - [ ] This comment contains a description of changes (with reason)
 - [ ] Referenced issue is linked
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] Documentation in `docs` is updated
 - [ ] `CHANGELOG.rst` is updated

**Description of changes**

* First draft setup from pip to poetry build system

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -8,11 +8,11 @@
...
- - [ ] This comment contains a description of changes (with reason)
- - [ ] Referenced issue is linked
- - [ ] If you've fixed a bug or added code that should be tested, add tests!
- - [ ] Documentation in `docs` is updated
- - [ ] `CHANGELOG.rst` is updated
+-   [ ] This comment contains a description of changes (with reason)
+-   [ ] Referenced issue is linked
+-   [ ] If you've fixed a bug or added code that should be tested, add tests!
+-   [ ] Documentation in `docs` is updated
+-   [ ] `CHANGELOG.rst` is updated
```
Comment (Zethson) [inline .github/pull_request_template.md:15]: Remove

File: .github/dependabot.yml [hunk 2]
```diff
@@ -1,78 +1,78 @@
...
-  - dependabot
-  commit-message:
-    prefix: "[DEPENDABOT]"
+    # cookietemple itself
+    - package-ecosystem: pip
+      directory: "/"
+      schedule:
+          interval: daily
+          time: "04:00"
+      open-pull-requests-limit: 99
```
Comment (Zethson) [inline .github/dependabot.yml:9]: Decrease for all templates to 3

File: .github/workflows/build_package.yml [hunk 3]
```diff
@@ -3,36 +3,34 @@ name: Build cookietemple Package
...
+                  python-version: ${{ matrix.python }}
+
+            - name: Install Poetry
+              run: |
+                  pip install poetry
+                  poetry --version
+
+            - name: Build package
+              run: |
+                  poetry build --ansi
```
Comment (Zethson) [inline .github/workflows/build_package.yml:30]: Should be one liner without |

**Review by Zethson — CHANGES_REQUESTED** [review]
I'll review soon

File: .github/workflows/create_cli_python.yml [hunk 4]
```diff
@@ -1,61 +1,64 @@
...
+            - name: Build Package
+              run: |
+                  cd explodingspringfield
+                  make install
+
+            - name: Run Tests
+              run: |
+                  mv explodingspringfield /tmp
+                  cd /tmp/explodingspringfield
+                  git init
```
Comment (Zethson) [inline .github/workflows/create_cli_python.yml:54]: Why is a git repository required? I think I knew once, but I forgot...

File: .github/workflows/create_gui_java.yml [hunk 5]
```diff
@@ -1,56 +1,59 @@
...
+            - uses: actions/checkout@v2
+              name: Check out source-code repository
+
+            - name: Setup Python
+              uses: actions/setup-python@v2.2.2
+              with:
+                  python-version: ${{ matrix.python }}
+
+            - name: Install Poetry
+              run: |
```
Comment (Zethson) [inline .github/workflows/create_gui_java.yml:31]: So I am now trying to always skip the `|` for one-liners, but add it if there are two lines or more.

File: .github/workflows/publish_package.yml [hunk 6]
```diff
@@ -1,31 +1,34 @@
-name: Publish cookietemple to PyPI
+name: Publish Cookietemple to PyPI
```
Comment (Zethson) [inline .github/workflows/publish_package.yml:1]: Why uppercase? Think we settled on lowercase absolutely everywhere for consistency

File: .github/workflows/publish_package.yml [hunk 7]
```diff
@@ -1,31 +1,34 @@
...
 
 on:
-  release:
-    types: [created]
+    release:
+        types: [created]
```
Comment (Zethson) [inline .github/workflows/publish_package.yml:5]: Must be `types: [published]`!

File: .github/workflows/run_tests.yml [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+              run: |
+                  pip install poetry
+                  poetry --version
+
+            - name: Install Nox
+              run: |
+                  pip install nox nox-poetry
+                  nox --version
+
+            - name: Install Rich
```
Comment (Zethson) [inline .github/workflows/run_tests.yml:83]: I forgot this again. Why are we explicitly installing rich?

File: pyproject.toml [hunk 9]
```diff
@@ -0,0 +1,77 @@
+[tool.poetry]
+name = "cookietemple"
+version = "1.3.2"  # <<COOKIETEMPLE_FORCE_BUMP>>
+description = "TODO"
```
Comment (Zethson) [inline pyproject.toml:4]: Fix this TODO and the ones below!

File: noxfile.py [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+    sys.exit(1)
+
+package = "cookietemple"
+python_versions = ["3.8", "3.9"]
+nox.options.sessions = (
+    "pre-commit",
+    "safety",
+    "mypy",
+    "tests",
+    # "xdoctest",
```
Comment (Zethson) [inline noxfile.py:25]: Well, fine with me.

File: noxfile.py [hunk 11]
```diff
@@ -0,0 +1,190 @@
...
+
+    session.run("sphinx-build", *args)
+
+
+@session(python=python_versions)
+def docs(session: Session) -> None:
+    """Build and serve the documentation with live reloading on file changes."""
+    args = session.posargs or ["--open-browser", "docs", "docs/_build"]
+    session.install(".")
+    session.install("sphinx", "sphinx-autobuild", "sphinx-click", "sphinx-rtd-theme")
```
Comment (Zethson) [inline noxfile.py:184]: Would be smarter if this installed the stuff inside the docs/requirements.txt file

Comment (Zethson) [reply noxfile.py:184]: But not important. Only if you want to.

File: makefiles/Linux.mk [hunk 12]
```diff
@@ -74,11 +74,10 @@ servedocs: docs ## compile the docs watching for changes
...
 release: dist ## package and upload a release
-	twine upload dist/*
+	poetry release
 
 dist: clean ## builds source and wheel package
-	python setup.py sdist
-	python setup.py bdist_wheel
+	poetry build
 
 install: clean ## install the package to the active Python's site-packages
```
Comment (Zethson) [inline makefiles/Linux.mk:82]: https://github.com/cookiejar/cookietemple/commit/ea4e89e9676b88e0adaf0530bdf936d31c5782fb

File: makefiles/Windows.mk [hunk 13]
```diff
@@ -74,11 +74,10 @@ servedocs: docs ## compile the docs watching for changes
...
 release: dist ## package and upload a release
-	twine upload dist\*
+	poetry release
 
 dist: clean ## builds source and wheel package
-	python setup.py sdist
-	python setup.py bdist_wheel
+	poetry build
 
 install: clean ## install the package to the active Python's site-packages
```
Comment (Zethson) [inline makefiles/Windows.mk:82]: https://github.com/cookiejar/cookietemple/commit/ea4e89e9676b88e0adaf0530bdf936d31c5782fb

**Review by Zethson — CHANGES_REQUESTED** [review]
Couple of minor things. You're definitely close.

File: .github/workflows/create_cli_python.yml (hunk 4 above)
Comment (Imipenem) [reply .github/workflows/create_cli_python.yml:54]: Nox can only run from within a git repo

File: .github/workflows/run_tests.yml (hunk 8 above)
Comment (Imipenem) [reply .github/workflows/run_tests.yml:83]: Rich is used to print the error message in our noxfile if nox or nox-poetry has not been found

**Pull request merged**